File: src/main/java/fr/khalypso/statesearchlib/model/exception/OperatorApplicationException.java
package fr.khalypso.statesearchlib.model.exception;

import fr.khalypso.statesearchlib.model.Operator;
import fr.khalypso.statesearchlib.model.State;

/**
 * Exception thrown when an {@link Operator} can not be applied to a {@link State}.
 */
public class OperatorApplicationException extends RuntimeException {

	public OperatorApplicationException() {
		super("The operator can not be applied to the current state");
	}
}

File: library/src/main/kotlin/fr/khalypso/statesearchlib/model/exception/OperatorApplicationException.kt
package fr.khalypso.statesearchlib.model.exception

import java.lang.RuntimeException

/**
 * Exception thrown when an [fr.khalypso.statesearchlib.model.Operator] can not be applied to a [fr.khalypso.statesearchlib.model.State].
 */
class OperatorApplicationException : RuntimeException("The operator can not be applied to the current state")
